lib/src/component_declaration/component_base.dart

```diff
@@ -503,9 +483,9 @@ typedef PropsModifier(Map props);
 ///
 /// > Note: Implements [MapViewMixin] instead of extending it so that the abstract [Props] declarations
 /// don't need a constructor. The generated implementations can mix that functionality in.
-abstract class UiProps extends MapBase
+abstract class UiProps<K, V> extends _OverReactMapBase<K, V>
```

Comment:
Is there a reason generic parameters were added to `UiProps`/`UiState` (and other places)? I'm concerned that that might introduce type complexity in props class inheritances (which can be problematic in dart2js; see https://github.com/dart-lang/sdk/issues/14729) without providing much benefit.